### PR TITLE
Updates for scala 2.12 compatibility

### PR DESCRIPTION
**Pull request by LucaCanali**
This PR adds a few updates and minor changes to 
- allow compiling with scala 2.12 
- running with the current (January 2019) Spark master
- update usage of logging  

This PR has been manually tested with a few sample runs of TPCDS and TPCH benchmark on a YARN cluster.

### CODE REVIEW

**Comment by npoggi** [comment]
From the first tests on our platform seems to work fine both on spark 3 and previous versions.  Will let running some other tests during the weekend.  @mengxr do you see any potential conflicts in the code changes to the ML benchmarks?

**Comment by npoggi** [comment]
From the first tests on our platform seems to work fine both on spark 3 and previous versions.  Will let running some other tests during the weekend.  @mengxr do you see any potential conflicts in the code changes to the ML benchmarks?

**Comment by npoggi** [comment]
@yhuai could you do a quick review of the code?

File: README.md [hunk 1]
```diff
@@ -34,7 +34,8 @@ The first run of `bin/run` will build the library.
 
 ## Build
 
-Use `sbt package` or `sbt assembly` to build the library jar.
+Use `sbt package` or `sbt assembly` to build the library jar.  
+Use `sbt +package` to build for scala 2.11 and 2.12.
```
[the remark this answers is not in the document]
Comment (LucaCanali) [reply README.md]: sbt +package builds with the scala versions listed in built.sbt' s crossScalaVersions, namely scala 2.11.12 and 2.12.8, while sbt package will build with the version specified by the parameter scalaVersion, i.e. 2.11.12

**Pull request merged**